File: tyama_codeeval102.rb
#!/usr/bin/ruby
require 'json'
puts $<.map{|line|
	n=0
	json=JSON.parse(line)['menu']['items']
	json.size.times{|i|if json[i]&&json[i]['label'] then n+=json[i]['id'] end}
	n
}

File: codeeval/tyama_codeeval102.go
//usr/bin/env go run $0 $@;exit
package main
import(
	"os"
	"bufio"
	"encoding/json"
	"fmt"
)
type Item struct{
	Id int
	Label *string
}
type Menu struct{
	Header *string
	Items []*Item
}
type Data struct{
	Menu Menu
}

func main(){
	reader:=bufio.NewReader(os.Stdin)
	for {
		s,err := reader.ReadString('\n')
		if err!=nil { break }
		var data Data
		err = json.Unmarshal([]byte(s),&data)
		r:=0
		if err==nil {
			for _,item := range data.Menu.Items {
				if item!=nil && item.Label!=nil { r+=item.Id }
			}
		}
		fmt.Println(r)
	}
}